tests/ui/truncate_with_drain.fixed

```diff
@@ -0,0 +1,282 @@
+#![allow(unused)]
+#![warn(clippy::truncate_with_drain)]
+
+use std::collections::VecDeque;
+
+fn vec_range() {
+    // Do not lint because iterator is assigned
+    let mut v = vec![1, 2, 3];
+    let iter = v.drain(1..v.len());
+
+    // Do not lint because iterator is used
+    let mut v = vec![1, 2, 3];
+    let n = v.drain(1..v.len()).count();
+
+    // Do not lint because iterator is assigned and used
+    let mut v = vec![1, 2, 3];
+    let iter = v.drain(1..v.len());
+    let n = iter.count();
+
+    // Do lint
+    let mut v = vec![1, 2, 3];
+    v.truncate(1);
```

Comment:
Wouldn't be better to include error markers to ensure a regression can be identified automatically?

Reply:
I'm not sure I know what are error markers, did you mean adding new tests or something?

Reply:
They explain what error should be raised, and where. See e.g. https://github.com/rust-lang/rust-clippy/blob/master/tests/ui/into_iter_without_iter.rs#L10

Reply:
Thank you, I'll look into that